src/expint.jl

```diff
@@ -207,10 +207,10 @@ En_safe_gamma_term(ν::Integer, z::Real) = (z ≥ 0 || isodd(ν) ? 1 : -1) * exp
 # https://functions.wolfram.com/GammaBetaErf/ExpIntegralE/10/0005/
 # returns the two terms from the above equation separately
 function En_cf_gamma(ν::Number, z::Number, n::Int=1000)
-    A = float(1 - ν)
-    B::typeof(A) = 1
-    Bprev::typeof(A) = 0
-    Aprev::typeof(A) = 1
+    A, z = map(float, promote(float(1 - ν), z))
```

Comment:
The `ν::Int` + `z::Float32` example would still be problematic. As in other places in SpecialFunctions, I think you'd rather want
```suggestion
    A, z = map(float, promote(1 - ν, z))
```

Reply:
thanks, my mistake